plugins/modules/zos_copy.py

```diff
@@ -1793,21 +1870,37 @@ def create_seq_dataset_from_file(
         dest (str) -- Name of the data set.
         force (bool) -- Whether to replace an existing data set.
         is_binary (bool) -- Whether the file has binary data.
+        asa_text (bool) -- Whether the file has ASA control characters.
         volume (str, optional) -- Volume where the data set should be.
     """
     src_size = os.stat(file).st_size
-    record_format = record_length = None
+    # record_format = record_length = None
+    record_format = None
+    # When dealing with ASA files, if copying from USS,
+    # the record length will need to be adjusted (we know it
+    # comes from USS because those flows don't send a
+    # value for record_length, while flows from source data
+    # sets do).
+    adjust_record_format = False
 
     # When src is a binary file, the module will use default attributes
     # for the data set, such as a record format of "VB".
     if not is_binary:
         record_format = "FB"
-        record_length = get_file_record_length(file)
+        if not record_length:
+            record_length = get_file_record_length(file)
+            adjust_record_format = True
```

Comment:
Is there any issue with `record_format = "FB"` where ASA needs FBA , I saw other code updates to FBA so maybe those later overwrite this `FB` but wanted to highlight the point.

Reply:
The record format gets overwritten inside `get_data_set_attributes` if the module is copying ASA text files.

Reply:
Thanks @rexemin , I should have followed the code but was aiming to get a quick turn around on this PR, I do see it updated there.